### PR TITLE
Align To Amp

### LINKED ISSUE #5: Move to apriltag function (for amp)
This should use similar logic to the AprilTagTrackDrive command (use apriltags to reset odometry, then use odometry to align with the amp).

The driver should have control over the robot's forward/backward direction.

**Potential Implementation:**
* **Set fieldOriented to TRUE**, otherwise you won't be able to line up left/right until the robot is aligned with the amp.
* Set requested robot heading to apriltag heading (use same logic as `AprilTagTrackDrive`
* Velocity PID to align robot left/right (**this will be field oriented forward/back, i.e. 'x' axis**)
* Driver joystick to drive robot towards/away from amp (**this will be field oriented left/right, i.e. 'y' axis**). Use either the x or y value returned from `getDriverInputs()` according to driver preference (just choose one for now).

### LINKED ISSUE #29: Determine shooter angle and RPM using distance from speaker
**How does this improve the robot or repo?**
* Makes shooting from non-preset locations possible.

**Feature Description**
* Use distance from speaker to calculate angle assuming the note flies in a linear path.
* RPM equation should be determined empirically (ask @relishthesoda). The RPM should be high enough for the piece to travel in a straight line (makes calculating the angle considerably easier).

**Potential Implementation**
* Create a command that spins up the shooter and adjusts the angle to the calculated values when running.

### LINKED ISSUE #42: Use SwerveController to calculate angular velocity (give it heading) in AprilTagTrackDrive
**How does this improve the robot or repo?**
* Easier to maintain only one heading PID.

**Improvement Description**
* Instead of using a separate angular velocity PID, send `tx` to SwerveController as the desired heading (see TeleopHeadingDrive for the implementation).

### LINKED ISSUE #60: Invert drive controls when on red alliance
**How does this improve the robot or repo?**
* You can drive on red alliance.

**Improvement Description**
* Invert drive velocity and heading when on red alliance, because the robot always uses the blue coordinate system.

### PR DESCRIPTION
**Pull request by UserC2**
Closes #42 
Closes #5 

Tested by WhyNot180 & UserC2 in simulation.

Need to test on actual robot.

As a result of #56, also closes #29.
Closes #60 too.

### CODE REVIEW

File: src/main/java/frc/robot/commands/AprilTagTrackDrive.java
Comment (WhyNot180) [inline src/main/java/frc/robot/commands/AprilTagTrackDrive.java:1]: This still needs a check for alliance colour.

**Review by WhyNot180 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/Limelight.java
Comment (UserC2) [inline src/main/java/frc/robot/subsystems/Limelight.java:1]: AprilTagTrackDrive target rotation should be in AprilTagTrackDrive, NOT in Limelight.

Comment (UserC2) [reply src/main/java/frc/robot/subsystems/Limelight.java:1]: Less magic necessary (check whether this angle is actually correct)

File: src/main/java/frc/robot/commands/AprilTagTrackDrive.java
Comment (UserC2) [inline src/main/java/frc/robot/commands/AprilTagTrackDrive.java:1]: File good, magic bad.

**Pull request merged**